### PR TITLE
document non-default stats retention

### PR DESCRIPTION
**Pull request by taroface**
Relates to DOC-4740.

### CODE REVIEW

**Comment by taroface** [comment]
@rytaft Thanks! Using your entire suggestion. 

Just one question to clarify; is a column made "non-default" essentially because it has "non-default" historical statistics that are not refreshed? Or is there an existing definition of "non-default" that I'm not familiar with?

**Review by kathancox — APPROVED** [review]
LGTM!

File: v22.2/cost-based-optimizer.md [hunk 1]
```diff
@@ -60,11 +59,17 @@ Statistics are refreshed in the following cases:
...
+
+#### Configure non-default statistics retention
 
-On the other hand, if a table has 1,500,000,000 rows, 20% of that, or 300,000,000 rows, would have to become stale before auto statistics collection was triggered. With a table this large, you would have to lower `sql.stats.automatic_collection.fraction_stale_rows` significantly to allow for regular stats collections. This can cause smaller tables to have stats collected much more frequently, because it is a global setting that affects automatic stats collection for all tables.
+By default, when CockroachDB refreshes statistics for a column, it deletes the old statistics for the column (while leaving the most recent 4-5 historical statistics). When CockroachDB refreshes statistics, it also deletes the statistics for any "non-default" column sets. Historical statistics on non-default column sets should not be retained indefinitely, because they will not be refreshed automatically and could cause the optimizer to choose a suboptimal plan if they become stale. Such non-default historical statistics may exist because columns were deleted or removed from an index, and are therefore no longer part of a multi-column statistic.
```
[the remark this answers is not in the document]
Comment (taroface) [reply v22.2/cost-based-optimizer.md]: Changed to "previous". Thanks for pointing this out!

**Comment by rytaft** [comment]
> Just one question to clarify; is a column made "non-default" essentially because it has "non-default" historical statistics that are not refreshed? Or is there an existing definition of "non-default" that I'm not familiar with?

The "default" statistics that are collected are:
- single-column statistics on every indexed column in the table, as well as 100 additional columns
- multi-column statistics on every index prefix. For example, given a table with columns `{a, b, c, d}` and an index on `(a, b, c)`, we will collect stats on `a`, `b`, `c`, `d`, `{a, b}`, and `{a, b, c}`.

Existing stats may become non-default if either a column or an index is dropped. For example, if column `d` is dropped and index `(a, b, c)` is also dropped, now the existing stats on `d`, `{a, b}`, and `{a, b, c}` are "non-default" because they will not be automatically refreshed. The stats on `a`, `b`, and `c` are the new default stats.

Does this help clarify?

**Comment by taroface** [comment]
Yes, that answers my question. Thanks a lot for explaining this!

**Pull request merged**